### PR TITLE
[7.x] Fix bad method call exception on LazyCollection dd() & dump() methods

### LINKED ISSUE #33926: dd() on Lazy collection returns bad method call exception: push does not exist.
- Laravel Version: 7.x
- PHP Version: 7.4
- Database Driver & Version:

### Description:
Calling the `dd()` on a lazy collection returns bad method call exception: push does not exist.
```php
BadMethodCallException
Method Illuminate\Support\LazyCollection::push does not exist.
```

### Steps To Reproduce:
Run this code:

```php
$lazyCollection = LazyCollection::times(INF)->tapEach(function ($value) {
        dump($value);
});

$lazyCollection->take(20)->dd();
```

### PR DESCRIPTION
**Pull request by stephenjude**
This PR fixes bad method call exception on LazyCollection dd() & dump() methods.

Fixes #33926

### CODE REVIEW

File: src/Illuminate/Support/LazyCollection.php [hunk 1]
```diff
@@ -36,6 +36,19 @@ public function __construct($source = null)
             $this->source = $this->getArrayableItems($source);
         }
     }
+    
+    /**
+     * Convert source to iterable
+     *
+     * @param $source
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/LazyCollection.php:43]: invalid syntax

File: src/Illuminate/Support/LazyCollection.php [hunk 2]
```diff
@@ -36,6 +36,19 @@ public function __construct($source = null)
...
+     * @return static
+     */
+    private function push($source)
+    {
+        return (new static(function () use ($source) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/LazyCollection.php:48]: Don't need extra brackets around this

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Support/LazyCollection.php (hunk 1 above)
Comment (GrahamCampbell) [reply src/Illuminate/Support/LazyCollection.php:43]: Still invalid syntax.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Illuminate/Support/LazyCollection.php (hunk 1 above)
Comment (stephenjude) [reply src/Illuminate/Support/LazyCollection.php:43]: I am a bit lost. How do I fix this?

Comment (mfn) [reply src/Illuminate/Support/LazyCollection.php:43]: It's `@param` followed by the **type** and then the variable name `$source`; see any other `@param` in the framework.

Comment (stephenjude) [reply src/Illuminate/Support/LazyCollection.php]: Okay, I just made a commit that fixed that. Thanks!

**Comment by taylorotwell** [comment]
Ping @JosephSilber

**Comment by JosephSilber** [comment]
`push` is a mutating method. [We have no mutating methods on the lazy collection](#29671).

I submitted a fix for the `dump` method here: #33944.

**Comment by stephenjude** [comment]
Okay.

**Pull request closed without merging**